search-app/src/cmr/search/results_handlers/csv_results_handler.clj

```diff
@@ -1,16 +1,83 @@
 (ns cmr.search.results-handlers.csv-results-handler
   "Handles the CSV results format and related functions."
-  (:require [cmr.common-app.services.search.elastic-results-to-query-results :as elastic-results]
-            [cmr.common-app.services.search.elastic-search-index :as elastic-search-index]
-            [cmr.common-app.services.search :as qs]
-            [cheshire.core :as json]
-            [clojure.data.csv :as csv]
-            [clojure.string :as str]
-            [cmr.search.services.acls.acl-results-handler-helper :as acl-rhh])
+  (:require
+   [cheshire.core :as json]
+   [clojure.data.csv :as csv]
+   [clojure.string :as str]
+   [cmr.common-app.services.search :as qs]
+   [cmr.common-app.services.search.elastic-results-to-query-results :as elastic-results]
+   [cmr.common-app.services.search.elastic-search-index :as elastic-search-index]
+   [cmr.search.services.acls.acl-results-handler-helper :as acl-rhh]
+   [cmr.search.services.query-execution.granule-counts-results-feature :as gcrf])
   (:import
     [java.io StringWriter]))
 
-(def CSV_HEADER
+(def COLLECTION_CSV_HEADER
+  ["Data Provider"
+   "Short Name"
+   "Version"
+   "Entry Title"
+   "Processing Level"
+   "Platforms"
+   "Start Time"
+   "End Time"])
+
+(defmethod elastic-search-index/concept-type+result-format->fields [:collection :csv]
+ [concept-type query]
+ (let [csv-fields ["provider-id"
+                   "short-name"
+                   "version-id"
+                   "entry-title"
+                   "processing-level-id"
+                   "platforms"
+                   "start-time"
+                   "end-time"]]
+   (distinct (concat csv-fields acl-rhh/collection-elastic-fields))))
+
+(defmethod elastic-results/elastic-result->query-result-item [:collection :csv]
+  [context query elastic-result]
+  (let [{start-date :start-date
+         end-date :end-date
+         provider-id :provider-id
+         processing-level :processing-level-id
+         entry-title :entry-title
+         platforms :platforms
+         version :version-id
+         short-name :short-name} (:_source elastic-result)
+        start-date (when start-date (str/replace (str start-date) #"\+0000" "Z"))
+        end-date (when end-date (str/replace (str end-date) #"\+0000" "Z"))
+        platform-short-names (->> platforms
+                                  (map :short-name)
+                                  (str/join ","))
+        collection-id (:_id elastic-result)]
+    (merge {:row [provider-id
+                  short-name
+                  (str version)
+                  entry-title
+                  (str processing-level)
+                  platform-short-names
+                  start-date
+                  end-date]
+            :id collection-id}
+           (acl-rhh/parse-elastic-item :collection elastic-result))))
+
+(defmethod gcrf/query-results->concept-ids :csv
```

Comment:
I don't think this function is used. We should remove it.

Reply:
hmm. On second thought, this function is needed, but the implementation should be raising an error saying `include_granule_counts` is not supported for csv format. We should have a test for `include_granule_counts` in csv format.

Reply:
The standard error message currently used by CMR for other formats is: 
e.g. KML:
`Collections search in kml format is not supported with include_granule_counts option`